erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -170,9 +170,10 @@ def set_missing_values(self):
 
 			if not self.party:
 				frappe.throw(_("Party is mandatory"))
-
-			_party_name = "title" if self.party_type in ("Student", "Shareholder") else self.party_type.lower() + "_name"
-			self.party_name = frappe.db.get_value(self.party_type, self.party, _party_name)
+			
+			if not self.party_name:
+				_party_name = "title" if self.party_type in ("Student", "Shareholder") else self.party_type.lower() + "_name"
+				self.party_name = frappe.db.get_value(self.party_type, self.party, _party_name)
```

Comment:
```suggestion
				field_name = "title" if self.party_type in ("Student", "Shareholder") else self.party_type.lower() + "_name"
				self.party_name = frappe.db.get_value(self.party_type, self.party, field_name)
```

Better variable name.